app/models/supporter.rb

```diff
@@ -135,12 +138,22 @@ def address_field_attributes
     attributes.slice(*ADDRESS_FIELDS)
   end
 
+  def filled_address_fields?
+    address_field_attributes.any? { |column, value| value.present? }
+  end
+
   def update_primary_address
     if self.changes.slice(*ADDRESS_FIELDS).any? #changed an address field
-      unless primary_address.nil?
-        primary_address.update(address_field_attributes)
-      else
-        self.addresses.build(address_field_attributes)
+      if filled_address_fields?
+        if primary_address.nil?
+          self.addresses.build(address_field_attributes)
+        else
+          primary_address.update(address_field_attributes)
+        end
+      elsif primary_address.present?
+        self.addresses.delete(primary_address)
```

Comment:
What happens when you run just this line and not the next two?

Reply:
If I run just `self.addresses.delete(primary_address)` it deletes the primary address from the list of addresses for this supporter, but the address continues to reference this supporter.